### ISSUE TITLE
Ability to scroll to top of body with modal / scrollLock

### ISSUE THREAD

**Issue by jonyo** [kobaltedev/kobalte#263]
**Is your feature request related to a problem? Please describe.**
We have a Kobalte dialog modal for filters on a search.

At some resolutions, the main body is still visible.

When changing the filters causes the search results to change, I want to scroll the body to the top (behind the modal).

I can use `window.scrollTo(...)` and that works fine for chrome, but Safari on iOS it does not work.

**Describe the solution you'd like**
Some exposed way to scroll the body to the top.

**Describe alternatives you've considered**
I set `window.document.body.top = '0px';` and that works while the modal is still open... But as soon as the modal closes it seems Kobalte scrolls to the position the body was at when the modal opened, which scrolls them back down again.

**Comment by jer3m01** [comment]
I don't have a device to test on and this seems more like a Safari issue, but does using:
```js
const scrollElm = document.scrollingElement;
scrollElm.scrollTop = 0;
```
instead of `window.scrollTo(...)` fix the issue?

https://developer.mozilla.org/en-US/docs/Web/API/document/scrollingElement

Another option would be to disable Kobalte's scroll lock with (`v0.11.3`):
```jsx
<Dialog.Root preventScroll={false}>...</Dialog.Root>
```
and handle the scroll position yourself.

Let me know if this work!

**Comment by jonyo** [comment]
Hi @jer3m01 thanks for the reply!

> preventScroll

We like the handy scroll lock most of the time, sometimes we want to change it though.

So when the dialog is closing, it does one last scroll:

https://github.com/kobaltedev/kobalte/blob/27f29a7538f0a6fd476be7e84d024e7e662edb46/packages/core/src/primitives/create-prevent-scroll/create-prevent-scroll.ts#L130

That happens after everything else, so you can't undo it right away. At least not without a timeout or something, which would not be a good user experience.

In other words, it isn't the scrolling that does not work... I'm able to make it scroll to the spot ok on safari as long as the modal is still open. The problem is that when the modal closes, that line above triggers, and it scrolls right back to where it was, undoing my manual "scroll to the top" I did earlier.

Hope that makes sense?

**Comment by jer3m01** [comment]
I've added a way to disable scrolling back, you'll have to control the open state (until #321) and disable default prevent scroll:

```tsx
const [open, setOpen] = createSignal(false);

createPreventScroll({
    enabled: open,
    scrollBack: false, // Disables call to `window.scrollTo()`
});

return (<Dialog.Root open={open} onOpenChange={setOpen} preventScroll={false}>...</Dialog.Root>);
```

**Comment by jer3m01** [comment]
Reopening this as I've added a dependency to `solid-prevent-scroll` instead of our custom fork.
This is missing the `scrollBack` option but from my small test this new package doesn't have this issue.

If you're still having unwanted behavior, please comment and I'll quickly add the option to the upstream dependency.
Otherwise let me know everything is good and I'll close this issue.
